plugins/scenario/hu.bme.mit.gamma.scenario.model/src/hu/bme/mit/gamma/scenario/util/ScenarioModelValidator.java

```diff
@@ -396,6 +405,72 @@ public Collection<ValidationResultMessage> checkScenarioReferenceParamCount(Scen
 		return validationResultMessages;
 	}
 
+	public Collection<ValidationResultMessage> checkScenarioCheck(ScenarioCheckExpression check) {
+		Collection<ValidationResultMessage> validationResultMessages = new ArrayList<ValidationResultMessage>();
+		List<EventParameterReferenceExpression> params = ecoreUtil.getAllContentsOfType(check.getExpression(),
+				EventParameterReferenceExpression.class);
+		if (!params.isEmpty()) {
+			EObject container = check.eContainer();
+			if (container instanceof ModalInteractionSet) {
+				ModalInteractionSet set = (ModalInteractionSet) container;
+				List<Signal> signals = set.getModalInteractions().stream().filter(it -> it instanceof Signal)
+						.map(it -> (Signal) it).collect(Collectors.toList());
+				for (EventParameterReferenceExpression paramRef : params) {
+					Event event = paramRef.getEvent();
+					Port port = paramRef.getPort();
+					boolean ok = false;
+					for (Signal signal : signals) {
+						if(signal.getPort().equals(port) && signal.getEvent().equals(event)) {
```

Comment:
Shouldn't you use `==` instead of `equals` here?

Reply:
Wouldn't that just compare the references themselves in java?